app/models/miq_report_result/result_set_operations.rb

```diff
@@ -0,0 +1,43 @@
+module MiqReportResult::ResultSetOperations
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    def apply_limit_and_offset(results, options)
+      results.slice(options['offset'].to_i, options['limit'].to_i) || []
+    end
+
+    def format_result_set(miq_report, result_set)
+      tz = miq_report.get_time_zone(Time.zone)
+
+      col_format_hash = miq_report.col_order.zip(miq_report.col_formats).to_h
+
+      result_set.map! do |row|
+        row.map do |key, _|
+          [key, miq_report.format_column(key, row, tz, col_format_hash[key])]
+        end.to_h
+      end
+    end
+
+    def filter_result_set(report, result_set, options)
```

Comment:
Yes you are right I was able to remove couple of usage of "map" method and I simplified the code.

> little confused that filter is formatting.

it is because we need to filter exaclty what user see - so we need to apply filter on formatted data. 

I did not throwing out formatted column which is used for filtering and now such column is skipped when I am formatting rest of columns.
algorithm with filtering:

```
input: result_set, filter_string, filter_column
output: filtered and formatted result_set (=output_result_set)
--------
1) formatted_result_set = format values only for filter_column in result_set
2) filtered_result_set = filter formatted_result_set according to filter_string on filter_column
3) output_result_set = format other columns expect to filter_column in filtered_result_set
```

related commits:
https://github.com/ManageIQ/manageiq/pull/18532/commits/e290ad10e1f136e888204a3f8914b60a49e048ca

https://github.com/ManageIQ/manageiq/pull/18532/commits/e79365382960642b23d487194208c43fa85929aa